marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/OpenAPI.hs

```diff
@@ -62,6 +72,111 @@ instance ToJSON OpenApiWithEmptySecurity where
 
 type API = "openapi.json" :> Get '[JSON] OpenApiWithEmptySecurity
 
+data OpenApiLintIssue = OpenApiLintIssue
+  { trace :: Text
+  , message :: Text
+  }
+  deriving (Show, Eq)
+
+lintOpenApi :: OpenApi -> [OpenApiLintIssue]
+lintOpenApi oa = schemaDefinitionLints <> pathParametersLints <> pathOperationLints
+  where
+    showStackTrace :: [Text] -> Text
+    showStackTrace = Text.concat . List.intersperse "/" . List.reverse
+
+    schemaDefinitions :: Definitions Schema
+    schemaDefinitions = Optics.view (components . schemas) oa
+
+    paramDefinitions :: Definitions Param
+    paramDefinitions = Optics.view (components . parameters) oa
+
+    schemaRef :: Referenced Schema -> Maybe Schema
+    schemaRef = \case
+      Inline s -> Just s
+      Ref (Reference ((`IOHM.lookup` schemaDefinitions) -> s)) -> s
+
+    paramRef :: Referenced Param -> Maybe Param
+    paramRef = \case
+      Inline s -> Just s
+      Ref (Reference ((`IOHM.lookup` paramDefinitions) -> s)) -> s
+
+    schemaRule1Check :: [Text] -> Schema -> [OpenApiLintIssue]
+    schemaRule1Check stackTrace s = do
```

Comment:
This should probably be recursive because a schema can look like this:
```json
{
  "required": ["myfield"],
  "oneof": { "oneof": { ... } }
}
```